common/src/main/java/com/microsoft/identity/common/internal/platform/AndroidDevicePopManager.java

```diff
@@ -217,7 +217,9 @@ private KeyPair generateNewRsaKeyPair(@androidx.annotation.NonNull final Context
                         trySetAttestationChallenge = false;
 
                         continue;
-                    } else if (tryStrongBox && Build.VERSION.RELEASE_OR_CODENAME.equals("UpsideDownCake") && isNegativeInternalError(e.getCause())) {
+                    } else if (tryStrongBox && (Build.VERSION.SDK_INT >= Build.VERSION_CODES.R)
+                            && (Build.VERSION.RELEASE_OR_CODENAME.equals("UpsideDownCake") || Build.VERSION.RELEASE_OR_CODENAME.equals("14"))
```

Comment:
Why do we need RELEASE_OR_CODENAME at all? Can't we just rely on VERSION.SDK_INT? I think that should cover all cases, no?

Reply:
I think it should be sufficient to use VERSION.SDK_INT unless I am missing something. @fadidurah  do we have any reason to use Build.VERSION.RELEASE_OR_CODENAME? I think why this was added previously was because we wanted to restrict this to the beta/development versions.. and find the right fix. But now that this is going to be applied to a release code name as well, we should use VERSION.SDK_INT instead.

Reply:
I believe in my initial investigation, VERSION.SDK_INT would return 33, most likely that was related to the Beta of android 14. I'm guessing this would change in Android 14 proper release. If we wanted to be safe, we can include that as well in the or statement.